examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -143,15 +143,12 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
 
     Messaging::SendFlags sendFlags;
     System::PacketBufferHandle payloadBuf;
-    char * requestData = nullptr;
+    char requestData[kMsgBufSize];
 
     uint32_t size = gPingArguments.GetEchoReqSize();
-    VerifyOrExit(size <= kMaxPayloadSize, err = CHIP_ERROR_MESSAGE_TOO_LONG);
+    VerifyOrExit(size >= kMsgBufSize, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
```

Comment:
I don't understand this check.  Why are we checking that `GetEchoReqSize()` will _not_ fit in our buffer?

Reply:
gPingArguments.GetEchoReqSize() is specified by cli, its size should not less than the content buffer size since the content within requestData is copied to packet buffer payload in NewWithData

Reply:
OK, but I'm still not following this.  We are asked to send an echo packet with `size` bytes in it, right?  We check that `size >= kMsgBufSize`.  We set up a buffer of size `kMsgBufSize`.  We try to put some bytes in the buffer (in practice somewhere between 15 and 34, depending on the value of `GetEchoCount()`, and probably closer to 15 than 34).  At this point `requestData` may contain some number of initialized bytes, which is not all of them.  Or maybe it's all of them.  We don't really know.

Then we do `MessagePacketBuffer::NewWithData(requestData, size)`.  This will try to copy `size` bytes out of `requestData` into the packetbuffer.  But we just tested that `size >= kMsgBufSize` so at best we get lucky and they are equal and then we are OK; otherwise we try to read random data off the stack, which seems like a bad idea.  Are there any tests for the situation where `size > kMsgBufSize`?  Do they test that the data that's sent is sane?

Reply:
Since we only care about the payload size, no need to set the initial content payload after switching to API New.